### ISSUE TITLE
Function versioning

### ISSUE THREAD

**Issue by sklink** [harris-chris/Jot.jl#38]
Apologies for continuing to bombard you here!

We're looking at a deploy process where we would publish versions of the Lambda function and point the API to the correct version when we're ready to make the switch.

This means new code would be built and uploaded by Jot, and we would manually publish the version in AWS.

When I ran the Jot deployment today, the previously deployed version on the function disappeared. I couldn't find the log indicating where the function is getting written within the Jot deploy process to check whether the current process is to purge.

We'll work around this by renaming the function in the code before deploying. Just wanted to check in to see whether you had a quick fix on hand.

Thanks!

**Comment by harris-chris** [comment]
Yes, your guess here is correct, creating a new function will overwrite any
function with the same name (effectively, Jot does not have any concept of
function versions). I don't think it's a quick fix to add it but it doesn't
sound like a very complex one either. The functionality to capture lambda
function output is about 60% done and I'm hoping to get that finished this
weekend. The other feature I'd like to add is to move to PackageCompiler v2
(and therefore Julia v1.8), so I don't want to give any guarantees as to
when I'd be able to add function versions as a feature.

**Comment by sklink** [comment]
No rush, thanks for putting time to this!
